app/code/Magento/Sitemap/Model/Sitemap.php

```diff
@@ -576,9 +576,9 @@ protected function _getSitemapRow($url, $lastmod = null, $changefreq = null, $pr
             foreach ($images->getCollection() as $image) {
                 $row .= '<image:image>';
                 $row .= '<image:loc>' . $this->_escaper->escapeUrl($image->getUrl()) . '</image:loc>';
-                $row .= '<image:title>' . $this->escapeXmlText($images->getTitle()) . '</image:title>';
+                $row .= '<image:title>' . $this->escapeXmlText((string)$images->getTitle()) . '</image:title>';
```

Comment:
I think it should be `$image` instead of `$images`
```suggestion
                $row .= '<image:title>' . $this->escapeXmlText((string)$image->getTitle()) . '</image:title>';
```

Reply:
@engcom-Hotel, I have analyzed and debugged the code and found that the current implementation is correct. The reason is that $image does not contain any title; the title is associated only with $images. This is why, in the current codebase, it is $images instead of $image. Please have a look at below screenshots:
<img width="1792" alt="image" src="https://github.com/user-attachments/assets/a015a6bf-232b-46bd-a46b-029f55e8e77f">
<img width="1792" alt="image" src="https://github.com/user-attachments/assets/3f9a45a2-9484-471e-88e8-c0a1c30e6f48">


Additionally, for the caption, we are already checking if it is null, so IMO, we don't need the typecasting, as the error mentioned is related to null values only.

Since the changes is related to typecasting only and we already have automated tests for this class, could you please review the implementation?

Thanks!

cc: @RPrudon